lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -197,9 +180,244 @@ def config_name(self) -> str:
         return self.value
 
 
-class ServiceManager:
+class ServiceManager(ABC):
     """Control a Slurm service."""
 
+    @abstractmethod
+    def __init__(self, service: ServiceType) -> None: ...
+
+    @abstractmethod
+    def enable(self) -> None:
+        """Enable service."""
+
+    @abstractmethod
+    def disable(self) -> None:
+        """Disable service."""
+
+    @abstractmethod
+    def restart(self) -> None:
+        """Restart service."""
+
+    @abstractmethod
+    def active(self) -> bool:
+        """Return True if the service is active."""
+
+    @property
+    @abstractmethod
+    def type(self) -> ServiceType:
+        """Return the service type of the managed service."""
+
+
+class MungeKeyManager(ABC):
+    """Control the munge key."""
+
+    @abstractmethod
+    def get(self) -> str:
+        """Get the current munge key.
+
+        Returns:
+            The current munge key as a base64-encoded string.
+        """
+
+    @abstractmethod
+    def set(self, key: str) -> None:
+        """Set a new munge key.
+
+        Args:
+            key: A new, base64-encoded munge key.
+        """
+
+    @abstractmethod
+    def generate(self) -> None:
+        """Generate a new, cryptographically secure munge key."""
+
+
+class _EnvManager:
+    """Control configuration of environment variables used in Slurm components."""
+
+    def __init__(self, file: Union[str, os.PathLike], prefix: str, keys: [str]) -> None:
+        self._file: Path = Path(file)
+        self._service = prefix
+        self._keys: [str] = keys
```

Comment:
I added this because I needed it 😅
Previously I tried to set some env variable for the charms (I think it was for the Mysql issue), but it didn't really work, and after minutes of debugging I just realized I had made a typo on the variable name 🤣

The good thing about doing the sanity check is that it makes silly errors like this noticeable, and not silently do the wrong thing

Reply:
Though, If we're exposing the configs as properties instead of exposing the env manager directly, I think we can safely remove the sanity checks, since the errors will be caught by our tests anyways.

Reply:
Removed the checks. Any errors should be caught by unit tests anyways.